test/loading.jl

```diff
@@ -4,10 +4,8 @@ using Base.Test
 
 @test @__LINE__ == 5
 
-@test_throws ArgumentError Core.include("test_sourcepath.jl\0")
-
 include("test_sourcepath.jl")
-thefname = "the fname!//\\&\0\1*"
+thefname = "the fname!//\\&\1*"
```

Comment:
The value was accepted but is basically treated as a C string up to the NUL character. It is never used anywhere or checked against anything. You can see this by doing something that actually uses the filename.

```jl
julia> thefname = "the fname!\0and this part is missing"
"the fname!\0and this part is missing"

julia> include_string("include_string_test() = error()", thefname)()
ERROR: 
 in include_string_test() at ./the fname!:1
 in eval(::Module, ::Any) at ./boot.jl:234
 in macro expansion at ./REPL.jl:92 [inlined]
 in (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:46
```

Note that the filename for `include_string_test` doesn't have anything after the `\0` byte.

The commit remove the support for non NUL-terminating string explicitly so passing a malformed string like that throws an error.